meson.build

```diff
@@ -0,0 +1,1207 @@
+endif
+if compiler.has_header('setjmp.h')
+  os_auto.set('PJ_HAS_SETJMP_H', 1)
+endif
+if compiler.has_header('stdarg.h')
+  os_auto.set('PJ_HAS_STDARG_H', 1)
+endif
+if compiler.has_header('stddef.h')
+  os_auto.set('PJ_HAS_STDDEF_H', 1)
+endif
+if compiler.has_header('stdio.h')
+  os_auto.set('PJ_HAS_STDIO_H', 1)
+endif
+if compiler.has_header('stdint.h')
+  os_auto.set('PJ_HAS_STDINT_H', 1)
+endif
+if compiler.has_header('stdlib.h')
+  os_auto.set('PJ_HAS_STDLIB_H', 1)
+endif
+if compiler.has_header('string.h')
+  os_auto.set('PJ_HAS_STRING_H', 1)
+endif
+if compiler.has_header('sys/ioctl.h')
+  os_auto.set('PJ_HAS_SYS_IOCTL_H', 1)
+endif
+if compiler.has_header('sys/select.h')
+  os_auto.set('PJ_HAS_SYS_SELECT_H', 1)
+endif
+if compiler.has_header('sys/socket.h')
+  os_auto.set('PJ_HAS_SYS_SOCKET_H', 1)
+endif
+if compiler.has_header('sys/time.h')
+  os_auto.set('PJ_HAS_SYS_TIME_H', 1)
+endif
+if compiler.has_header('sys/timeb.h')
+  os_auto.set('PJ_HAS_SYS_TIMEB_H', 1)
+endif
+#define PJ_HAS_SYS_TYPES_H 1
+#define PJ_HAS_SYS_FILIO_H 1
+#define PJ_HAS_SYS_SOCKIO_H 1
+#define PJ_HAS_SYS_UTSNAME_H 1
+if compiler.has_header('time.h')
+  os_auto.set('PJ_HAS_TIME_H', 1)
+endif
+if compiler.has_header('unistd.h')
+  os_auto.set('PJ_HAS_UNISTD_H', 1)
+endif
+if compiler.has_header('execinfo.h')
+  os_auto.set('PJ_HAS_EXECINFO_H', 1)
+endif
+#undef PJ_HAS_MSWSOCK_H */
+#undef PJ_HAS_WINSOCK_H */
+#undef PJ_HAS_WINSOCK2_H */
+#undef PJ_HAS_WS2TCPIP_H */
+#define PJ_SOCK_HAS_IPV6_V6ONLY 1
+#define PJ_SOCK_HAS_INET_ATON 1
+#define PJ_SOCK_HAS_INET_PTON 1
+#define PJ_SOCK_HAS_INET_NTOP 1
+#define PJ_SOCK_HAS_GETADDRINFO 1
+#define PJ_SOCK_HAS_SOCKETPAIR 1
+#undef PJ_HAS_PTHREAD_MUTEXATTR_SETTYPE */
+#undef PJ_PTHREAD_MUTEXATTR_T_HAS_RECURSIVE */
+#define PJ_SOCKADDR_HAS_LEN 1
+os_auto.set('PJ_HAS_SOCKLEN_T', 1)
+#define PJ_SELECT_NEEDS_NFDS 0
+#undef PJ_HAS_LINUX_EPOLL */
+os_auto.set('PJ_HAS_ERRNO_VAR', 1)
+os_auto.set('PJ_HAS_SO_ERROR', 1)
+os_auto.set('PJ_BLOCKING_ERROR_VAL', 'EAGAIN')
+os_auto.set('PJ_BLOCKING_CONNECT_ERROR_VAL', 'EINPROGRESS')
+os_auto.set('PJ_HAS_THREADS', 1)
+os_auto.set('PJ_HAS_HIGH_RES_TIMER', 1)
+#define PJ_HAS_MALLOC 1
+#   define PJ_OS_HAS_CHECK_STACK    0
+#define PJ_HAS_LOCALTIME_R 1
+#define PJ_NATIVE_STRING_IS_UNICODE 0
+#define PJ_POOL_ALIGNMENT 8
+os_auto.set('PJ_ATOMIC_VALUE_TYPE', 'long')
+if os == 'darwin'
+  os_auto.set('PJ_GETHOSTIP_DISABLE_LOCAL_RESOLUTION', 1)
+  os_auto.set('PJ_SOCKADDR_USE_GETADDRINFO', 1)
+endif
+os_auto.set('PJ_EMULATE_RWMUTEX', 0)
+os_auto.set('PJ_THREAD_SET_STACK_SIZE', 0)
+os_auto.set('PJ_THREAD_ALLOCATE_STACK', 0)
+os_auto.set('PJ_HAS_SSL_SOCK', 1)
+os_auto.set('PJ_SSL_SOCK_IMP', 'PJ_SSL_SOCK_IMP_OPENSSL')
+#undef PJ_HAS_PTHREAD_NP_H */
+#define PJ_HAS_PTHREAD_SETNAME_NP 1
+#undef PJ_HAS_PTHREAD_SET_NAME_NP */
+#
+configure_file(output : 'os_auto.h', configuration : os_auto)
+run_command('/bin/cp', 'builddir/os_auto.h', 'pjlib/include/pj/compat/os_auto.h', check : true)
```

Comment:
Why?

Reply:
I did not understand how meson can publish the file in the source code. Recreating the directory tree in the `builddir` seemed like a very time consuming task for me. So I chose a simple copy here. What do you suggest?

Reply:
I'm not certain I understand what you mean by "publish", as it will always be recreated from meson when you setup a build directory.

- Why does this file need to be in the source directory? You can include it just fine from the build directory.
- How do you know the build directory is called "builddir"? What if people use `meson setup builddir-debug/` and `meson setup builddir-release/` and possibly also use `builddir-clang/` or `builddir-gcc/`? This is why meson has `meson.current_build_dir()`.

I would probably use `configure_file()` from inside `pjlib/include/pj/compat/meson.build`. A C/C++ include will find that file from anywhere on the compiler search path, it doesn't have to be in the same directory tree as other files in `#include <pj/compat/*.h>`.

Reply:
> Why does this file need to be in the source directory?

See article https://trac.pjsip.org/repos/wiki/Getting-Started/Build-Preparation

> How do you know the build directory is called "builddir"? What if people use ...

My version `meson-sh` use that directory. Quick answer is people make refer changes ...